### PR TITLE
Minor refactoring inside RestPostRepository

### PR DESCRIPTION
**Pull request by markusrubey**

### CODE REVIEW

File: data/src/main/java/dk/nodes/template/data/network/RestPostRepository.kt [hunk 1]
```diff
@@ -7,38 +7,31 @@ import kotlinx.coroutines.channels.BroadcastChannel
...
 
 class RestPostRepository @Inject constructor(private val api: Api) : PostRepository {
 
-    private val postsChannel = BroadcastChannel<List<Post>>(Channel.CONFLATED)
-        .also {
-            it.offer(listOf())
-        }
+    private val postsChannel = BroadcastChannel<List<Post>>(Channel.CONFLATED).apply {
+        offer(listOf())
+    }
```
Comment (lucassales2) [inline data/src/main/java/dk/nodes/template/data/network/RestPostRepository.kt]: Can be reduced even more replacing it with` ConflatedBroadcastChannel<List<Post>>(listOf())`

Comment (markusrubey) [reply data/src/main/java/dk/nodes/template/data/network/RestPostRepository.kt]: Sweet!

File: data/src/main/java/dk/nodes/template/data/network/RestPostRepository.kt [hunk 2]
```diff
@@ -7,38 +7,31 @@ import kotlinx.coroutines.channels.BroadcastChannel
...
-                response.errorBody()?.string(),
-                response.message()
-            ))
+    override suspend fun getPosts(): List<Post> = with(api.getPosts()) {
+        return result.also {
+            postsChannel.send(it)
         }
     }
+
+    private val <T> Response<T>.result: T
```
Comment (lucassales2) [inline data/src/main/java/dk/nodes/template/data/network/RestPostRepository.kt]: I'm not sure about this, throwing an exception on a field should not be the expected behavior

Comment (markusrubey) [reply data/src/main/java/dk/nodes/template/data/network/RestPostRepository.kt]: agree 🙈, maybe just have it as a function?

```kotlin
    @Throws(RepositoryException::class)
    private fun <T> Response<T>.result(): T = body()?.let {
        return it
    } ?: throw(RepositoryException(
            code = code(),
            errorBody = errorBody()?.string(),
            msg = message()
    ))
```

Comment (markusrubey) [reply data/src/main/java/dk/nodes/template/data/network/RestPostRepository.kt]: prolly should also be available to other remote repository impls

Comment (levinzonr) [reply data/src/main/java/dk/nodes/template/data/network/RestPostRepository.kt:26]: I'd call it `getResultOrThrow()` then

**Pull request closed without merging**